src/plugins/data/public/query/query_string/language_service/lib/language_reference.test.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render, screen, fireEvent, cleanup, act, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { LanguageReference } from './language_reference';
+import { IntlProvider } from 'react-intl';
+
+// Helper to wrap component with IntlProvider
+const renderWithIntl = (ui) => {
+  return render(<IntlProvider locale="en">{ui}</IntlProvider>);
+};
+
+describe('LanguageReference Component', () => {
+  beforeEach(() => {
+    // Clear localStorage and DOM before each test
+    localStorage.clear();
+    document.body.innerHTML = '';
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  test('auto-opens the info box on first load if autoShow is true and sets localStorage key', async () => {
```

Comment:
Not a blocker, but can you add a fast follow test to ensure that this only opens once. Thats the key behavior here. Lets not forget to test that.

Reply:
Sure lets add that case as a followup